### ISSUE TITLE
bug: DeepSeek-OCR greedy decode emits the first token 128 times (step logits discarded)

### ISSUE THREAD

**Issue by tobocop2** [kreuzberg-dev/kreuzberg#1210]
## Problem

DeepSeek-OCR's greedy decode loop never consumes the logits it computes after the first token. The loop calls `forward_step` each iteration but throws the result away (`let _ = ...`) and re-reads the initial forward's logits every time, so the "generated" text is the first token repeated MAX_NEW_TOKENS times. On the hello-world fixture the output is literally `Hello` concatenated 128 times:

```
HelloHelloHelloHelloHelloHello...  (640 chars, 128 tokens, no stop)
```

The e2e test's degenerate-repeat guard cannot see this because it splits on whitespace and the repeats have none. A second, masked bug sits underneath: the step offset restarts near zero (`step + 1`) instead of continuing at the token's absolute position, so RoPE positions for generated tokens are wrong even once the logits are wired through.

Reproduced against main @ 293f1eaadc on a RunPod A100 host with the released weights.

## Suggested fix

Make the loop feed each step's logits into the next iteration and pass the token's absolute position (prompt length + step) as the offset. With both fixed the model reads the full "Hello World" off the fixture and stops at EOS in a fraction of the time (68s vs 360s of wasted identical steps).

Fix is in #1205.
